apim-apk-agent/pkg/transformer/transformer.go

```diff
@@ -0,0 +1,429 @@
+		return "", 0, fmt.Errorf("error while generating the apk yaml")
+	}
+	return string(c), apiYamlData.RevisionID, nil
+}
+
+// getAPIType will be selecting the appropriate API type need to be added in the apk-conf
+// based on the type mentioned in the api.json
+func getAPIType(protocolType string) string {
+	var apiType string
+	switch protocolType {
+	case "HTTP", "HTTPS":
+		apiType = "REST"
+	case "GraphQL":
+		apiType = "GraphQL"
+	}
+	return apiType
+}
+
+// Generate the interceptor policy if request or response policy exists
+func getReqAndResInterceptors(reqPolicyCount int, resPolicyCount int) (*[]OperationPolicy, *[]OperationPolicy) {
+	var reqInterceptor, resInterceptor []OperationPolicy
+	var interceptorParams *InterceptorService
+	var interceptorPolicy OperationPolicy
+
+	if reqPolicyCount > 0 || resPolicyCount > 0 {
+		interceptorParams = &InterceptorService{
+			BackendURL:      "https://interceptor-svc.ns:9081",
+			HeadersEnabled:  true,
+			BodyEnabled:     true,
+			TrailersEnabled: true,
+			ContextEnabled:  true,
+			TLSSecretName:   "interceptor-cert",
+			TLSSecretKey:    "ca.crt",
+		}
+
+		// Create an instance of OperationPolicy
+		interceptorPolicy = OperationPolicy{
+			PolicyName:    "Interceptor",
+			PolicyVersion: "v1",
+			Parameters:    interceptorParams,
+		}
+	}
+
+	if reqPolicyCount > 0 {
+		reqInterceptor = append(reqInterceptor, interceptorPolicy)
+	}
+
+	if resPolicyCount > 0 {
+		resInterceptor = append(resInterceptor, interceptorPolicy)
+
+	}
+
+	return &reqInterceptor, &resInterceptor
+}
+
+// GenerateUpdatedCRs takes the .apk-conf, api definition, vHost and the organization for a particular API and then generate and returns
+// the relavant CRD set as a zip
+func GenerateUpdatedCRs(apkConf string, apiDefinition string, vHosts []string, organizations []string) (*bytes.Buffer, error) {
+	if apkConf == "" {
+		return nil, errors.New("Error: APK-Conf can't be empty")
+	}
+
+	if apiDefinition == "" {
+		return nil, errors.New("Error: API Definition can't be empty")
+	}
+
+	// Create a buffer to store the request body
+	var requestBody bytes.Buffer
+	writer := multipart.NewWriter(&requestBody)
+
+	// Add apkConfiguration field and store the passed APK Conf file
+	apkConfField, err := writer.CreateFormField("apkConfiguration")
+	if err != nil {
+		panic(err)
+	}
+	_, err = io.Copy(apkConfField, strings.NewReader(apkConf))
+	if err != nil {
+		panic(err)
+	}
+
+	// Add apiDefinition field and store the passed API Definition file
+	definitionField, err := writer.CreateFormField("definitionFile")
+	if err != nil {
+		panic(err)
+	}
+	_, err = io.Copy(definitionField, strings.NewReader(apiDefinition))
+	if err != nil {
+		panic(err)
+	}
+
+	// Close the multipart writer
+	writer.Close()
+
+	//configDSHost := config.ReadConfig().APIM.ConfigDeployerHost
+	configDSHost := "https://api.am.wso2.com:9095/api/configurator/1.0.0/apis"
```

Comment:
shall we provide this configuration as a parameter for this method, by reading the config values

Reply:
Modified the code to read the config and get the endpoint